etcdctl/ctlv3/command/watch_command.go

```diff
@@ -101,27 +101,33 @@ func watchInteractiveFunc(cmd *cobra.Command, osArgs []string, envKey, envRange
 		l = strings.TrimSuffix(l, "\n")
 
 		args := argify(l)
-		if len(args) < 2 && envKey == "" {
-			fmt.Fprintf(os.Stderr, "Invalid command %s (command type or key is not provided)\n", l)
-			continue
-		}
+		switch args[0] {
+		case "watch":
+			if len(args) < 2 && envKey == "" {
+				fmt.Fprintf(os.Stderr, "Invalid command %s (command type or key is not provided)\n", l)
+				continue
+			}
+			watchArgs, execArgs, perr := parseWatchArgs(osArgs, args, envKey, envRange, true)
+			if perr != nil {
+				ExitWithError(ExitBadArgs, perr)
+			}
 
-		if args[0] != "watch" {
+			ch, err := getWatchChan(c, watchArgs)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Invalid command %s (%v)\n", l, err)
+				continue
+			}
+			go printWatchCh(c, ch, execArgs)
+		case "progress":
```

Comment:
Q. Does this work? Think watch streams are not shared in etcdctl? So, this progress request would be no-op?

Reply:
Hm. I'm not familiar with that aspect of etcdctl.  Here's a sample run I just did:
```
ETCDCTL_API=3 bin/etcdctl watch --interactive
watch a
progress
progress notify: 14
watch b
progress
progress notify: 14
```

Is there a case you can think of that might exercise a problem with the separate watch streams?

Reply:
@jpbetz Oh, actually you are right. This is for interactive watch command, so requests are sharing the stream. I was thinking of just regular watch command, and a separate progress command.

Reply:
@gyuho I can work on adding more edge cases to the `TestWatchRequestProgress` clientv3/integration test that I added in this PR.  I currently just does:  (1) start a watch (2) put a k/v (3) verify the k/v is recieved over the watch (4) request progress (5) check that progress is received over the watch.

Corner cases might be:
- No watches
- More than one watch
- Put a key not visible to the watch, ensure progress is incremented

If you think of any others, let me know.

Reply:
I modified the watch stream publisher to send the progress notify responses on all watch channels, the result is etcdctl now does:

```
ETCDCTL_API=3 bin/etcdctl watch --interactive
watch a
progress
progress notify: 14
watch b
progress
progress notify: 14
progress notify: 14
```

Reply:
@jpbetz I tried those corner cases with this branch, and confirm that it's covered. I can't think of any, for now.
Thanks!